flight/PiOS.posix/posix/pios_led.c

```diff
@@ -28,19 +28,53 @@
 
 /* Project Includes */
 #include "pios.h"
+#include "openpilot.h"
 
 #if defined(PIOS_INCLUDE_LED)
 
 /* Private Function Prototypes */
 
+DONT_BUILD_IF(PIOS_LED_NUM > 8, piosLedNumTooHigh);
 
 /* Local Variables */
-static uint8_t LED_GPIO[PIOS_LED_NUM];
+static uint8_t ledStatus;
 
+static inline void PIOS_SetLED(uint32_t led, uint8_t stat) {
+	uint8_t newStatus = ledStatus;
 
-static inline void PIOS_SetLED(uint32_t LED,uint8_t stat) {
-	printf("PIOS: LED %i status %i\n",LED,stat);
-	LED_GPIO[LED]=stat;
+	PIOS_Assert(led < PIOS_LED_NUM);
```

Comment:
FWIW, this was a nice indication that the simulator was alive and kicking. Maybe there's a way to keep this behavior without spamming the console, e.g. a message that periodically says "status LED blinked 100 times".